src/_pytest/main.py

```diff
@@ -249,7 +253,11 @@ def pytest_cmdline_main(config):
     return wrap_session(config, _main)
 
 
-def _main(config, session):
+def _main(
+    config: Config, session: "Session"
+) -> Optional[
+    Union["Literal"[ExitCode.TESTS_FAILED], "Literal"[ExitCode.NO_TESTS_COLLECTED]]
```

Comment:
Until pyflakes has a new release, it'd be better to put the entire thing in quotes, then the `noqa` on the `Literal` import is not needed.

You can also combine the literals.

So: `"Union[Literal[ExitCode.TESTS_FAILED, ExitCode.NO_TESTS_COLLECTED]]"`.

Reply:
Without `Union` then even?  `Optional["Literal[ExitCode.TESTS_FAILED, ExitCode.NO_TESTS_COLLECTED]"]`

But `noqa` is still required.

Reply:
Right. I think for the noqa to go away you need to wrap the entire type in quotes, including the `Optional`. Really not important though.

Reply:
Also did not work IIRC.
The problem here is that pyflakes does not look at the strings to begin with.
I think it is fine like it is now.